packages/fluentui/react-context-selector/src/createContext.ts

```diff
@@ -1,33 +1,35 @@
 import * as React from 'react';
-import { Context, ContextListener, ContextValue, CreateContextOptions } from './types';
 
-// Stops React Context propagation
-// https://github.com/facebook/react/blob/95bd7aad7daa80c381faa3215c80b0906ab5ead5/packages/react-reconciler/src/ReactFiberBeginWork.js#L2656
-const calculateChangedBits = () => 0;
+import { Context, ContextValue } from './types';
+import { runWithNormalPriority, useIsomorphicLayoutEffect } from './utils';
 
 const createProvider = <Value>(Original: React.Provider<ContextValue<Value>>) => {
   const Provider: React.FC<React.ProviderProps<Value>> = props => {
-    const listeners = React.useRef<ContextListener<Value>[]>([]);
-
-    // We call listeners in render intentionally. Listeners are not technically pure, but
-    // otherwise we can't get benefits from concurrent mode.
-    //
-    // We make sure to work with double or more invocation of listeners.
-    listeners.current.forEach(listener => listener(props.value));
-
-    // Disables updates propogation for React Context as `value` is always shallow equal
-    const subscribe = React.useCallback((listener: ContextListener<Value>) => {
-      listeners.current.push(listener);
-
-      const unsubscribe = () => {
-        const index = listeners.current.indexOf(listener);
-        listeners.current.splice(index, 1);
+    // TODO: write better descriptions
+    const valueRef = React.useRef(props.value); // value from props.value
+    const versionRef = React.useRef(0); // render/effect counter
+    const contextValue = React.useRef<ContextValue<Value>>(); // stable object to avoid context updates
```

Comment:
```suggestion
    // Holds an actual "props.value"
    const valueRef = React.useRef(props.value);
    // Used to sync context updates and avoid stale values, can be considered as render/effect counter of Provider.
    const versionRef = React.useRef(0);
    
    // A stable object, is used to avoid context updates via mutation of its values.
    const contextValue = React.useRef<ContextValue<Value>>();
```